lib/standalone.js

```diff
@@ -76,14 +91,28 @@ module.exports = function (options/*: Object */)/*: Promise<stylelint$standalone
       }
     }
 
-    const getStylelintResults = filePaths.map(filePath => {
+    let absoluteFilePaths = filePaths.map(filePath => {
       const absoluteFilepath = (!path.isAbsolute(filePath))
         ? path.join(process.cwd(), filePath)
         : filePath
+
+      return absoluteFilepath
+    })
+
+    if (useCache) {
+      absoluteFilePaths = absoluteFilePaths.filter(fileCache.hasFileChanged.bind(fileCache))
+    }
+
+    const getStylelintResults = absoluteFilePaths.map(absoluteFilepath => {
+      debug(`Processing ${absoluteFilepath}`)
       return stylelint._lintSource({
         filePath: absoluteFilepath,
       }).then(postcssResult => {
-        return stylelint._createStylelintResult(postcssResult, filePath)
+        if (postcssResult.stylelint.stylelintError && useCache) {
+          debug(`File has problems, skip caching: ${absoluteFilepath}`)
```

Comment:
Let's be more clear with this message. How about `${absoluteFilepath} contains linting errors and will not be cached.`?